contracts/integrations/test/exchange/transaction_protocol_fee_test.ts

```diff
@@ -0,0 +1,420 @@
+
+            before(async () => {
+                let newOrder = await maker.signOrderAsync();
+                let data = exchangeDataEncoder.encodeOrdersToExchangeData(ExchangeFunctionName.FillOrder, [newOrder]);
+                transactionB2 = await bob.signTransactionAsync({ data });
+
+                newOrder = await maker.signOrderAsync();
+                data = exchangeDataEncoder.encodeOrdersToExchangeData(ExchangeFunctionName.FillOrder, [newOrder]);
+                transactionC2 = await charlie.signTransactionAsync({ data });
+
+                const transactions = [transactionA, transactionB, transactionC];
+                const signatures = transactions.map(tx => tx.signature);
+                const recursiveData = await deployment.exchange
+                    .batchExecuteTransactions(transactions, signatures)
+                    .getABIEncodedTransactionData();
+                nestedTransaction = await alice.signTransactionAsync({ data: recursiveData });
+            });
+
+            it('Alice batchExecuteTransactions => nested batchExecuteTransactions, Bob fillOrder, Charlie fillOrder; mixed protocol fees', async () => {
+                const transactions = [nestedTransaction, transactionB2, transactionC2];
+                const signatures = transactions.map(tx => tx.signature);
+                const txReceipt = await deployment.exchange
+                    .batchExecuteTransactions(transactions, signatures)
+                    .awaitTransactionSuccessAsync({ from: alice.address, value: MIXED_FEES_WITH_REFUND });
+                expectedBalances.simulateFills(
+                    [order, order, order, order, order],
+                    [alice.address, bob.address, charlie.address, bob.address, charlie.address],
+                    txReceipt,
+                    deployment,
+                    MIXED_FEES_WITH_REFUND,
+                );
+            });
+            it('Alice batchExecuteTransactions => Bob fillOrder, nested batchExecuteTransactions, Charlie fillOrder; mixed protocol fees', async () => {
+                const transactions = [transactionB2, nestedTransaction, transactionC2];
+                const signatures = transactions.map(tx => tx.signature);
+                const txReceipt = await deployment.exchange
+                    .batchExecuteTransactions(transactions, signatures)
+                    .awaitTransactionSuccessAsync({ from: alice.address, value: MIXED_FEES_WITH_REFUND });
+                expectedBalances.simulateFills(
+                    [order, order, order, order, order],
+                    [bob.address, alice.address, bob.address, charlie.address, charlie.address],
+                    txReceipt,
+                    deployment,
+                    MIXED_FEES_WITH_REFUND,
+                );
+            });
+            it('Alice batchExecuteTransactions => Bob fillOrder, Charlie fillOrder, nested batchExecuteTransactions; mixed protocol fees', async () => {
```

Comment:
This is great - can you have a `batchExecuteTransaction` nested in an `executeTransaction`?